shell/common/extensions/api/tabs.json

```diff
@@ -129,15 +146,14 @@
             "type": "integer",
             "name": "tabId",
             "minimum": 0
-          },
-          {
-            "type": "function",
-            "name": "callback",
-            "parameters": [
-              {"name": "tab", "$ref": "Tab"}
-            ]
           }
-        ]
+        ],
+        "returns_async": {
+          "name": "callback",
+          "parameters": [
+            {"name": "tab", "$ref": "Tab"}
+          ]
+        }
```

Comment:
Regarding backporting, I'm not sure this will work cleanly as these schemas were only added late last year. Would need to check which Electron version makes the cutoff.

`returns_async` added 2020-11-05
https://source.chromium.org/chromium/chromium/src/+/6a8c35314cd95efb0d218f03d38af6c5fadb03c9

Promises support on the tabs API added 2020-11-12
https://source.chromium.org/chromium/chromium/src/+/83a34182e0fc3dc3e3948da775f4a9178e66bd1b

Reply:
Thanks, I will remember this when backporting